app/routes/v1/gql.js

```diff
@@ -0,0 +1,239 @@
+    orgId,
+  };
+  sendReqToGraphql({ req, res, query, variables, operationName });
+}));
+
+router.post('/channels/:uuid/versions', getOrgId, asyncHandler(async(req, res)=>{
+  // #swagger.tags = ['channels']
+  // #swagger.summary = 'Adds a new channel version'
+  const { orgId } = req;
+  const operationName  = 'addChannelVersion';
+  const query = `
+    mutation ${operationName}($orgId:  String!, $channelUuid: String!, $name: String!, $type: String!, $content: String, ) {
+      addChannelVersion(orgId: $orgId, channelUuid: $channelUuid, name: $name, type: $type, content: $content ){
+        versionUuid
+        success
+      }
+    }
+  `;
+  const channelUuid = req.params.uuid;
+  const name = req.body.name;
+  const type = req.body.type;
+  const content = req.body.content;
+
+  if(!name || !channelUuid || !type || !content){
+    throw new Error('needs { channelUuid, name, type, content }');
+  }
+  const variables = {
+    orgId,
+    channelUuid,
+    name,
+    type,
+    content,
+  };
+  sendReqToGraphql({ req, res, query, variables, operationName });
+}));
+
+router.get('/clusters/clustersByOrgId', getOrgId, asyncHandler(async(req, res)=>{
+  // #swagger.tags = ['clusters']
+  // #swagger.summary = 'Gets all clusters for an org'
+  const { orgId } = req;
+  const operationName  = 'clustersByOrgId';
+  const query = `
+    query ${operationName}($orgId: String!) {
+      clustersByOrgId(orgId: $orgId) {
+        id
+        orgId
+        clusterId
+        metadata
+      }
+    }
+
+  `;
+  const variables = {
+    orgId,
+  };
+  sendReqToGraphql({ req, res, query, variables, operationName });
+}));
+
+router.post('/groups', getOrgId, asyncHandler(async(req, res)=>{
+  // #swagger.tags = ['groups']
+  // #swagger.summary = 'Adds a group'
+  const { orgId } = req;
+  const operationName  = 'addGroup';
+  const query = `
+    mutation ${operationName}($orgId:  String!, $name: String!) {
+      addGroup(orgId: $orgId, name: $name) {
+        uuid
+      }
+    }
+  `;
+  const name = req.body.name;
+  if(!name){
+    throw new Error('needs { name }');
+  }
+  const variables = {
+    orgId,
+    name,
+  };
+  sendReqToGraphql({ req, res, query, variables, operationName });
+}));
+
+router.put('/groups/groupClusters', getOrgId, asyncHandler(async(req, res)=>{
+  // #swagger.tags = ['groups']
+  // #swagger.summary = 'Assigns a group to a cluster'
+  const { orgId } = req;
+  const operationName  = 'groupClusters';
+  const query = `
+    mutation ${operationName}($orgId:  String!, $uuid: String!, $clusters: [String!]!) {
+      groupClusters(orgId: $orgId, uuid: $uuid, clusters: $clusters) {
+        modified
+      }
+    }
+  `;
+  const uuid = req.body.uuid;
+  const clusters = req.body.clusters;
+  if(!uuid || !clusters){
+    throw new Error('needs { uuid, clusters }');
+  }
+  const variables = {
+    orgId,
+    uuid,
+    clusters,
+  };
+  sendReqToGraphql({ req, res, query, variables, operationName });
+}));
```

Comment:
`/groups/groupClusters` is not proper syntax for a REST request.  Instead, client should be able to:
- POST `/groups` to create a group, specifying the name _and the clusters_ in the body
- PUT `/groups/[uuid]` to update a group, specifying the new clusters array in the body

If the graphql API doesnt allow creating a group with the clusters list, and the clusters list must be updated separately, it may even be necessary to make _two_ graphql calls for a single `POST`.

Reply:
Fixed by latest commits, other than allowing specifying clusters in the POST to `/groups`.  That will require more work, perhaps ensuring multiple graphql queries can be sent without returning a response to the client (one to create the group, one to set the clusters, and only sending back 201 CREATED response if both succeeded, else 400 or similar).

At this time I would propose including detailed code comments at least explaining that only the name can be specified on the POST to /groups and why it cannot include clusters.

Reply:
latest commit added code comment explaining current limitation.